cupy/core/_routines_statistics.pyx

```diff
@@ -173,6 +177,7 @@ __device__ min_max_st<T> my_argmax_complex(
     if (is_nan(b.value.imag())) return b;
     return (a.value >= b.value) ? a : b;
 }
+
```

Comment:
This fix is not needed.

Reply:
Sorry I didn't understand this one. Should I not remove the previous my_nanargmin and my_nanargmax_float templates? Because the rest is as it is I suppose.

Reply:
I have added more test cases hopefully resolves it now.

Reply:
Please review and suggest any changes